src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala

```diff
@@ -152,9 +153,13 @@ object FilterIndexRule
         val indexManager = Hyperspace
           .getContext(spark)
           .indexCollectionManager
-        val candidateIndexes = RuleUtils.getCandidateIndexes(indexManager, r)
 
-        candidateIndexes.filter { index =>
+        // TODO: the following check only considers indexes in ACTIVE state for usage. Update
+        //  the code to support indexes in transitioning states as well.
+        //  See https://github.com/microsoft/hyperspace/issues/65
+        val allIndexes = indexManager.getIndexes(Seq(Constants.States.ACTIVE))
+
+        val candidateIndexes = allIndexes.filter { index =>
```

Comment:
nit: Can you add a brief comment here as why the order of code blocks matter here (first index schema check then signature computation/comparison - similar to what you already explained in the PR description)? This is to make sure nobody in a future PR moves the code blocks around and causes negative perf impact.

Reply:
Thanks added some comment below. :)